Dockerfile

```diff
@@ -1,42 +1,53 @@
-FROM debian:bookworm-slim AS builder
+FROM alpine:3.20 AS builder
 
-RUN apt-get update && \
-    apt-get install -y \
+RUN apk add --update --no-cache \
         binutils-dev \
-        build-essential \
+        # Debian: build-essential
+        build-base \
         cmake \
         git \
-        libcurl4-openssl-dev \
-        libdw-dev \
-        libiberty-dev \
-        libssl-dev \
+        # Debian: libcurl4-openssl-dev
+        curl-dev \
+        curl-static \
+        # Debian: libdw-dev
+        libdw \
+        # Debian, alpine bundled in binutils-dev
+        # libiberty-dev \
+        # Debian: libssl-dev
+        openssl-dev \
         ninja-build \
         python3 \
-        zlib1g-dev \
-        ;
+        # Debian: zlib1g-dev
+        zlib-dev \
+        # Debian: libelf-dev
+        elfutils-dev \
+        # Debian: libstdc++-12-dev
+        libstdc++-dev \
+    ;
 
 ADD . /src/
 
 RUN mkdir /src/build && \
     cd /src/build && \
-    cmake -G 'Ninja' .. && \
+    export PATH="$PATH:/usr/lib/ninja-build/bin/" && \
+    CXXFLAGS="-D__ptrace_request=int" cmake -G 'Ninja' .. && \
     cmake --build . && \
     cmake --build . --target install
 
-FROM debian:bookworm-slim
+FROM alpine:3.20
 
 COPY --from=builder /usr/local/bin/kcov* /usr/local/bin/
 COPY --from=builder /usr/local/share/doc/kcov /usr/local/share/doc/kcov
 
 RUN set -eux; \
-    apt-get update; \
-    apt-get install -y --no-install-recommends --no-install-suggests \
-        libcurl4 \
-        libdw1 \
-        zlib1g \
+    apk add --update --no-cache \
+        bash \
+        libcurl \
+        libdw \
+        zlib \
+        libgcc \
+        libstdc++ \
     ; \
-    apt-get clean; \
-    rm -rf /var/lib/apt/lists/*; \
     # Write a test script
     echo '#!/usr/bin/env bash\nif [[ true ]]; then\necho "Hello, kcov!"\nfi' > /tmp/test-executable.sh; \
     # Test kcov
```

Comment:
You can remove the --include-pattern here. Since it's only one file, it's not really needed.

Since it's a bash script, you could consider adding `--bash-dont-parse-binary-dir` though, so that it doesn't try to find other files to cover in /tmp/

Reply:
None of the two options do work, what could be wrong on Alpine ?

Reply:
Since it's bash coverage, is bash actually installed in the image?

Most likely explanation is otherwise that there is some incompatibility in the version used in the docker image (i.e., most likely a newer bash version in Alpine).

Reply:
An uncommon option which disables the xtrace file descriptor is:

```--configure=bash-force-stderr-input=1```

not sure, but maybe could be the cause of the alpine issues. There's also `--bash-method=DEBUG`, but that sounds less likely.

Reply:
> Since it's bash coverage, is bash actually installed in the image?

Yes, I added a line or the script exec failed.

Okay, pushing `--configure=bash-force-stderr-input=1` to try

Reply:
Sorry, was looking in the source code.

`--debug-force-bash-stderr` should be the option. If that doesn't work, then perhaps we should let Alpine be?

However, would be nice to build docker images for riscv as well, which I also saw that you added  here (even if it's for bookworm)

Reply:
```json
 {
  "files": [
     {"file": "/tmp/coverage/bash-helper-debug-trap.sh", "percent_covered": "0.00", "covered_lines": "0", "total_lines": "3"},
     {"file": "/tmp/coverage/bash-helper.sh", "percent_covered": "0.00", "covered_lines": "0", "total_lines": "2"}
   ],
   "percent_covered": "0.00",
   "covered_lines": 0,
   "total_lines": 5,
   "percent_low": 25,
   "percent_high": 75,
  "command": "test-executable.sh",
  "date": "2024-07-08 10:50:40"
 }
```

Reply:
I tried pasting the commands locally and tested.

I wonder if  something is wrong there: At least I get no line endings with the command, which gives the same results as you see. If I manually edit the file, it works correctly:

```
Simons-iMac:emilpro ska$ cat /tmp/kcov/test-executable.sh/coverage.json
{
  "files": [

  ],
  "percent_covered": "0.00",
  "covered_lines": 0,
  "total_lines": 0,
  "percent_low": 25,
  "percent_high": 75,
  "command": "test-executable.sh",
  "date": "2024-07-08 13:11:52"
}
Simons-iMac:emilpro ska$ cat /tmp/test-executable.sh
#!/usr/bin/env bash\nif [[ true ]]; then\necho "Hello, kcov!"\nfi

[... edit the executable to get correct line endings ]

Simons-iMac:emilpro ska$ kcov --bash-dont-parse-binary-dir /tmp/kcov /tmp/test-executable.sh
Hello, kcov!
Simons-iMac:emilpro ska$ cat /tmp/test-executable.sh
#!/usr/bin/env bash
if [[ true ]]; then
echo "Hello, kcov!"
fi
Simons-iMac:emilpro ska$ cat /tmp/test-executable.sh
#!/usr/bin/env bash
if [[ true ]]; then
echo "Hello, kcov!"
fi
Simons-iMac:emilpro ska$ cat /tmp/kcov/test-executable.sh/coverage.json
{
  "files": [
    {"file": "/private/tmp/test-executable.sh", "percent_covered": "100.00", "covered_lines": "2", "total_lines": "2"}
  ],
  "percent_covered": "100.00",
  "covered_lines": 2,
  "total_lines": 2,
  "percent_low": 25,
  "percent_high": 75,
  "command": "test-executable.sh",
  "date": "2024-07-08 13:12:20"
}
```

Reply:
`--include-pattern` is needed or coverage drops

Reply:
Yes, that's fine. I think it's due to it finding other files in /tmp/ though, so `--bash-dont-parse-binary-dir` should also give 100%.

But go with --include-pattern, it's clear enough!